app/src/main/java/com/example/hellorecycler/MyAdapter.kt

```diff
@@ -3,57 +3,61 @@ package com.example.hellorecycler
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.ImageView
 import android.widget.TextView
 import androidx.recyclerview.widget.RecyclerView
+import kotlinx.android.synthetic.main.candy_text_layout.view.*
 
-// private val myDataset: Array<String>??
+// private val myDataset: Array<String>?? i get it now that the primary constructor creates the property
+// no additional declaration needed
 
-class MyAdapter(private val myDataset: List<String>) :
+class MyAdapter(private val myCandies: List<CandyData>, val clickListener: (CandyData)) :
```

Comment:
This constructor should show an error, right? Not sure how it wouldn't because you have to expicitly define the return type for a function like this.

you need to define your function like this: 

`val candyFunction : (CandyData) -> Unit`

also, I would name it something slightly different than click listener to help keep confusion cleared. This is not a listener, just a function/callback that is hit as a result of the listener being fired.

Reply:
thanks. will update